### PR TITLE
add e2e test framework

### PR DESCRIPTION
**Pull request by cofyc**
fixes https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/5 https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/4

This PR adds e2e test framework against Kubernetees cluster. It supports to run e2e tests against

- existing cluster started by hack/local-up-cluster.sh
- new cluster started by hack/local-up-cluster.sh
- new cluster started with GCE provider locally and in prow environment

prow job PR: https://github.com/kubernetes/test-infra/pull/10577

- 26b207e5c147b09991592d6fa011e847aeca553a: vendor only
- 322b09c23ad5c8d7be36dc0778ced49de7acec9e: test code
- bf358a5420f4298a2c0bf0cbe5d83ae498eefa8b: test entry point, etc 

This PR did not migrate all local volume provider e2e tests from k/k repo. I plan to migrate the rest of e2e tests and simplify code after e2e test workflow is set up.

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @msau42
Please take a look.

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+	immediateMode = storagev1.VolumeBindingImmediate
+)
+
+type localVolumeType string
+
+const (
+	// default local volume type, aka a directory
+	DirectoryLocalVolumeType localVolumeType = "dir"
+	// like DirectoryLocalVolumeType but it's a symbolic link to directory
+	DirectoryLinkLocalVolumeType localVolumeType = "dir-link"
```
Comment (msau42) [inline test/e2e/e2e_test.go]: are all off these constants needed?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+kubernetes/
+kubernetes.tar.gz
```
Comment (msau42) [inline .dockerignore]: Does this test actually download and build kubernetes?

Comment (cofyc) [reply .dockerignore]: In prow job, it downloads (extract ci/lastest by default) and start multiple node cluster with GCE provider. It does not build kubernetes but provisioner is built under this directory, ignoring these files speeds up docker build. This is for performance optimization.

Comment (cofyc) [reply .dockerignore]: It is weird at first glance, I added some comments.

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (cofyc) [reply test/e2e/e2e_test.go]: Cleaned.

**Pull request merged**